File: sdks/go/corthan/billing.go
package corthan

import (
	"context"
	"net/http"
)

// BillingService wraps Billing operations.
type BillingService struct {
	t *TransportClient
}

// GetTier retrieves billing tier details and Aurora resource usage metering.
func (s *BillingService) GetTier(ctx context.Context) (*BillingTierSuccess, error) {
	var result BillingTierSuccess
	err := s.t.Request(ctx, http.MethodGet, "/billing/tier", nil, &result)
	if err != nil {
		return nil, err
	}
	return &result, nil
}

File: sdks/node/src/services/billing.ts
import { TransportClient } from "../transport";
import { BillingTierSuccess } from "../types";

/**
 * Service for querying billing tier limits and usage meters.
 */
export class BillingService {
  constructor(private transport: TransportClient) {}

  /**
   * Retrieves active billing tier details and quota usage rates.
   */
  public async getTier(): Promise<BillingTierSuccess> {
    return this.transport.request<BillingTierSuccess>("GET", "/billing/tier");
  }
}
